src/iter/repeat.rs

```diff
@@ -0,0 +1,126 @@
+use super::*;
+use super::internal::*;
+use std::usize;
+
+/// Iterator adapter for [the `repeat()` function](fn.split.html).
+pub struct Repeat<T: Clone + Send> {
+    element: T,
+}
+
+/// Creates a parallel iterator that endlessly repeats `elt` (by
+/// cloning it). Note that this iterator has "infinite" length, so
+/// typically you would want to use `zip` or `take` or some other
+/// means to shorten it.
+///
+/// Example:
+///
+/// ```
+/// use rayon::prelude::*;
+/// use rayon::iter::repeat;
+/// let x: Vec<(i32, i32)> = repeat(22).zip(0..3).collect();
+/// assert_eq!(x, vec![(22, 0), (22, 1), (22, 2)]);
+/// ```
+pub fn repeat<T: Clone + Send>(elt: T) -> Repeat<T> {
+    Repeat { element: elt }
+}
+
+impl<T> ParallelIterator for Repeat<T>
+    where T: Clone + Send
+{
+    type Item = T;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge_unindexed(self, consumer)
+    }
+}
+
+impl<T> IndexedParallelIterator for Repeat<T>
+    where T: Clone + Send
+{
+    fn drive<C>(self, consumer: C) -> C::Result
+        where C: Consumer<Self::Item>
+    {
+        bridge(self, consumer)
+    }
+
+    fn with_producer<CB>(self, callback: CB) -> CB::Output
+        where CB: ProducerCallback<Self::Item>
+    {
+        callback.callback(RepeatProducer { repeat: self.element })
+    }
+
+    fn len(&mut self) -> usize {
+        usize::MAX
+    }
+}
+
+
+/// Producer
+
+struct RepeatProducer<T: Clone + Send> {
+    repeat: T,
+}
+
+impl<T: Clone + Send> Producer for RepeatProducer<T> {
+    type Item = T;
+    type IntoIter = RepeatIter<T>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        RepeatIter { repeat: self.repeat }
+    }
+
+    fn split_at(self, index: usize) -> (Self, Self) {
+        (RepeatProducer { repeat: self.repeat.clone() }, RepeatProducer { repeat: self.repeat })
```

Comment:
Ah yes, good point. On this note, I'm softening on `repeatn`. Seems like a useful building block.